ahnlich/server/src/main.rs

```diff
@@ -5,10 +5,85 @@ mod engine;
 mod errors;
 mod network;
 mod storage;
+use std::error::Error;
 
-#[cfg(test)]
-mod tests;
+use clap::{ArgAction, Args, Parser, Subcommand};
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::net::{TcpListener, TcpStream};
+
+async fn run_server() -> std::io::Result<()> {
+    let listener = TcpListener::bind("127.0.0.1:1369").await?;
 
-fn main() {
-    println!("Hello, world!");
+    loop {
+        let (stream, connect_addr) = listener.accept().await?;
+        println!("Connecting to {}", connect_addr);
+        tokio::spawn(async move {
+            if let Err(e) = process_stream(stream).await {
+                eprintln!("Error handling connection: {}", e)
+            };
+        });
+    }
 }
+
+async fn process_stream(stream: TcpStream) -> Result<(), tokio::io::Error> {
+    stream.readable().await?;
+    let mut reader = BufReader::new(stream);
+    loop {
+        let mut message = String::new();
+        let _ = reader.read_line(&mut message).await?;
```

Comment:
Nice.... this would be where we subsequently then deserialize the message coming in (as bytes) using bincode